src/structs/multiplexer.rs

```diff
@@ -118,6 +138,7 @@ where
 
     my_scope(|s| -> Result<()> {
         let (tx, rx) = channel();
+        let arc_processor = Arc::new(Box::new(processor));
```

Comment:
For my knowledge - why do you need Arc here, why you did not need it beforehand?

Reply:
Previously I used scoped threads, which control the lifetime of objects within the scope. However, now that the threads run in parallel during the lifetime of the context, we need a static lifetime which means that the data needs to be reference counted.